### ISSUE TITLE
Proposal: Make Configuration API stable

### ISSUE THREAD

**Issue by shivamkm07** [dapr/dapr#5522]
# Configuration API as Stable

This is the main tracking issue for moving Configuration API to stable. It follows the lifecycle process template defined [here](https://github.com/dapr/proposals/blob/main/templates/lifecycle.md).

> Note: Different task items for Alpha/Beta/Stable have been merged into Stable as Config API would be graduating directly to Stable as an exception once the items are cleared.

## Links
Links to any relevant resources go here:
* API design proposal: https://github.com/dapr/dapr/issues/2988
* Another related issue: https://github.com/dapr/dapr/issues/4684
 
# Acceptance criteria for Stable
> For each of the stages, add any specific tasks that need to be completed before this feature reaches that particular stage. If a particular item is *not needed* then a reason should be given.

## Target for v1.10
- [x] #6358
   * https://github.com/dapr/dapr/issues/5183
   * https://github.com/dapr/dapr/issues/4546
   * https://github.com/dapr/components-contrib/issues/2187
- [x] Minimum of one core SDK supports this feature (.NET / Python / Go / Java)
- [x] Feature documentation added to `dapr/docs`
- [x] Issue opened in `dapr/quickstarts` for quickstart examples to be created
- [x] Both HTTP and gRPC protocols implemented
- [x] HTTP API documentation added to the `Reference` section of Dapr documentation
- [x] No major changes to the API have occurred in the last XXX time period (releases? months?)
- [x] Interfaces to be used by `dapr/dapr` code defined and agreed upon
- [x] New building block package is defined in `components-contrib` repo
- [x] Minimum of two implementations of this building block
- [x] Telemetry data (metrics) available for this feature
- [x] SDK Spec is Updated
    * https://github.com/dapr/sig-sdk-spec/pull/23
    * https://github.com/dapr/sig-sdk-spec/pull/25
- [x] Quickstarts have been created
   * https://github.com/dapr/quickstarts/pull/743
 - [x] Conformance tests added for Config API testing one component
    * https://github.com/dapr/components-contrib/pull/2387
- [x] Conformance tests exercise both positive and negative cases
- [x] Certification tests added for one component
  * https://github.com/dapr/components-contrib/pull/2431
 
## Target for v1.11
- [x] Bug Fixes/Features
   * Update API from v1.0-alpha1 to v1.0, remove alpha1 in goSDK
      * https://github.com/dapr/dapr/pull/6251
      * https://github.com/dapr/dapr/pull/6272
      * https://github.com/dapr/components-contrib/pull/2807
- [x] E2E tests are up-to-date and comprehensive
  * https://github.com/dapr/dapr/pull/6128
- [ ] E2E tests exercise at least two different implementations of the building block's API
   * https://github.com/dapr/dapr/pull/6231
- [ ] Support in core SDKs (get, subscribe, unsubscribe)
   - [x] Python
      * https://github.com/dapr/python-sdk/pull/558
      * https://github.com/dapr/python-sdk/pull/557
   - [ ] Go
      * https://github.com/dapr/go-sdk/pull/389
      * https://github.com/dapr/go-sdk/pull/391
   - [ ] Java
      * https://github.com/dapr/java-sdk/pull/846
   - [ ] .NET
      * https://github.com/dapr/dotnet-sdk/pull/1077
   - [ ] JS
- [ ] Sign off from SDK maintainers on SDK implementations following SDK spec
- [x] APIs that are used in the building block also meet Stable criteria
- [x] Conformance tests validate at least two different components
   * https://github.com/dapr/components-contrib/pull/2740
- [x] Certification tests exist for at least two different components
    * https://github.com/dapr/components-contrib/pull/2744
    * https://github.com/dapr/components-contrib/pull/2825
- [x] Performance tests exist and regressions will prevent them from successful passing
   * https://github.com/dapr/dapr/pull/6247
- [ ] Performance data added to documentation (https://docs.dapr.io/operations/performance-and-scalability/)
- [ ] Documentation up-to-date with any new changes since Alpha
   * https://github.com/dapr/docs/issues/3342
#### P2 Issues (not a blocker for graduating Config API to stable)
  *  https://github.com/dapr/dapr/issues/5424

**Comment by chielos24** [comment]
Using this with our solution. Keyvault and Azure App config.

**Comment by mukundansundar** [comment]
retain v1-alpha1 and the alpha1 suffix -- as long as there are no breaking changes ... 
@shivamkm07 check if there are any breaking changes between 1.10 and master ...

**Comment by daixiang0** [comment]
Agree.
